cmd/prometheus/main.go

```diff
@@ -527,6 +527,9 @@ func main() {
 		conntrack.DialWithTracing(),
 	)
 
+	// This is passed to ruleManager.Update().
+	var externalURL = cfg.web.ExternalURL.String()
```

Comment:
why do we need a variable?

Reply:
The web flags aren't in the standard config object, they're actually in a separate `flagConfig` object. The reloader function, which is the entry point for the alerts, only takes the standard config, as do all the reloader functions. So, if we wanted to pass it the `flagConfig`, we'd have to change all the other reloader configs to take it too, even though they don't use it, and move the `flagConfg` to a separate package in order to prevent an import cycle. Setting a variable that can be accessed by the reloader function is just easier.

https://github.com/prometheus/prometheus/blob/51e4e2a5264172412506e5cb2e6c26507bd16c56/cmd/prometheus/main.go#L583-L599